bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -316,9 +383,12 @@ public String getCategoryIdentifier(SlingHttpServletRequest request) {
             return identifier;
         }
 
-        Map<String, String> categoryIdentifiers = categoryPageUrlFormat.parse(request.getRequestPathInfo());
+        Map<String, String> categoryIdentifiers = categoryPageUrlFormat
+            .parse(request.getRequestPathInfo(), request.getRequestParameterMap());
 
-        if (categoryIdentifiers.containsKey(URL_KEY_PARAM)) {
+        if (categoryIdentifiers.containsKey(UID_PARAM)) {
```

Comment:
This will re-add support for UIDs in the URL, do we really want this?

Reply:
If a customer implements a custom UrlFormat they should have all possible choices. In Magento it makes not much sense to use the uid, but if you implement a custom commerce backend and map it to something human readable it may make sense.

I think it is ok to "re-add" it ... not for the default configuration but the SPI.

Reply:
Ok to me. I agree on the 3rd party case. To have products & category in sync we should also support UID option for products.

Reply:
Makes sense but the ProductFilterInput does not provide an option to filter products by uid

https://devdocs.magento.com/guides/v2.4/graphql/queries/products.html#ProductFilterInput

So if it is used it may only be used for formatting a url. Also we already have the SKU which allows to have an identifier in the url that does not require a query to retrieve it from the backend.

Reply:
>Makes sense but the ProductFilterInput does not provide an option to filter products by uid

Ah right.